### ISSUE TITLE
Labs Select throws an a11y violation when using aria-required

### ISSUE THREAD

**Issue by jamesfan** [Workday/canvas-kit#781]
## 🐛 Bug Report

The Labs Select throws an a11y violation with [this test suite](https://ghe.megaleo.com/UIC/uic-common-utils/tree/master/utilities/a11y) when provided with an `aria-required` prop. The Labs Select doesn't perform any special handling for the `aria-required` prop, so it's spread onto the `button` rendered by the component along with the rest of the `elemProps`. However, `aria-required` is not an allowed attribute for a `button` according to the tests run by the referenced suite.

Workday Accessibility suggested we use a `button` for the Labs Select. [This example](https://www.w3.org/TR/wai-aria-practices-1.1/examples/listbox/listbox-collapsible.html) referenced on the [WAI-ARIA Authoring Practices](https://www.w3.org/TR/wai-aria-practices-1.1/#examples-6) also utilizes a `button`.

## To Reproduce

Follow the installation and usage instructions for the `uic-common-utils-a11y` tests: https://ghe.megaleo.com/UIC/uic-common-utils/tree/master/utilities/a11y

Run the `toHaveNoViolations()` test with `<Select aria-required={true}>` and you should be presented with something similar to the following violation:

```
{
  "id": "aria-allowed-attr",
  "impact": "critical",
  "tags": [
    "cat.aria",
    "wcag2a",
    "wcag412"
  ],
  "description": "Ensures ARIA attributes are allowed for an element's role",
  "help": "Elements must only use allowed ARIA attributes",
  "helpUrl": "https://dequeuniversity.com/rules/worldspace/3.5/aria-allowed-attr?application=attestAPI",
  "nodes": [
    {
      "any": [
        {
          "id": "aria-allowed-attr",
          "data": [
            "aria-required=\"true\""
          ],
          "relatedNodes": [],
          "impact": "critical",
          "message": "ARIA attribute is not allowed: aria-required=\"true\""
        }
      ],
      "all": [],
      "none": [],
      "impact": "critical",
      "html": "<button aria-haspopup=\"listbox\" aria-required=\"true\" type=\"button\" value=\"email\" class=\"css-abphv6 e6ydbno0\">E-mail</button>",
      "target": [
        "button"
      ],
      "failureSummary": "Fix any of the following:\n  ARIA attribute is not allowed: aria-required=\"true\""
    }
  ]
}
```

**Comment by jamesfan** [comment]
I'm currently blocked on guidance from Accessibility on how to handle this.

There isn't much information on `aria-required` online. The [WAI-ARIA Authoring Practices](https://www.w3.org/TR/wai-aria-practices-1.1/) make no mention of it. I was able to find this article on the topic: https://www.deque.com/blog/aria-requiredtrue-wcag-2-compliance-practice/

According to [MDN](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-required_attribute) and [Digital A11Y](https://www.digitala11y.com/combobox-role/), `aria-required` is permitted to be on the ARIA `combobox` role. We don't explicitly assign a role to the Select `button`, although Chrome automatically assigns it a role of `combobox`. Unfortunately, assigning the `combobox` role explicitly to the `button` throws a number of additional violations.

**Comment by jamesfan** [comment]
After discussing with Workday Accessibility, we've arrived at a solution: apply `aria-required` to the `listbox` rather than the `button`. Select consumers will also need to ensure the Select `label` includes the text "required" to ensure a11y users know the Select is required prior to opening the menu (`aria-required` will only notify them after the menu has been opened).
